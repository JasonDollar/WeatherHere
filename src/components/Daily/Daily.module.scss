.Daily {
  width: 100%;
  margin: 0 auto;
  overflow-x: hidden;
}

.Summary {
  font-size: 2.5rem;
  text-align: center;
  margin: 2rem 0 1rem 0;
}

.chartContainer {
  overflow-y: hidden;
  // background-color: var(--color-back);
}

@media only screen and (max-width: 576px) {
  .chartContainer {
    overflow-y: hidden;
    // margin: 0 1rem 0 -3rem;
  }

}