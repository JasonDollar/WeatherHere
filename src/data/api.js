const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

