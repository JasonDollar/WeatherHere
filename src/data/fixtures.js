import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}