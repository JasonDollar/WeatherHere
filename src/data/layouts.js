const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



