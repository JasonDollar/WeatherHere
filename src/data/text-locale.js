const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 