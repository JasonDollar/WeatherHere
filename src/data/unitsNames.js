export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}