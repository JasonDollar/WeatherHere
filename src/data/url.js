const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_URL }