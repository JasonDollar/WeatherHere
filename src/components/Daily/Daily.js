import React from 'react'
import PropTypes from 'prop-types'
import {
  BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer,
} from 'recharts'
import CustomTooltip from '../CustomTooltip/CustomTooltip'

import { getTimeFromSeconds, formatNumber } from '../../data/utils'
import classes from './Daily.module.scss'

const Daily = ({ daily, dateText, timezone }) => {
  // const test = daily.data.map(item => getTimeFromSeconds(item.time * 1000))

  const data = daily.data.map((item, index) => {
    
    const day = getTimeFromSeconds(item.time, timezone)
    let dayName
    if (index === 0) {
      dayName = dateText.today
    } else dayName = dateText.weekDay[day.day]
    return {
      name: dayName,
      // change temperature later to localized name
      [dateText.temperature]: [formatNumber(item.temperatureHigh), formatNumber(item.temperatureLow)],
      
    }
  })
  
  const style = '#dd0055'

  
  
  return (
    <div className={classes.Daily}>
      <div className={classes.Summary}>
        {daily.summary}
      </div>
      <ResponsiveContainer height={300} width="95%" margin={0}>
        <BarChart maxBarSize={60} data={data}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="name" />
          <YAxis label="°C" type="number" />
          <Tooltip
            content={<CustomTooltip tempText={dateText} type="daily" />}
            wrapperStyle={{
              backgroundColor: '#fff',
              border: '1px solid #555',
              borderRadius: '5px',
              boxShadow: '1px 1px 2px #333',
            }}
          />
          <Legend />
          <Bar dataKey={dateText.temperature} fill={style} unit="°C" />
          
        </BarChart>
      </ResponsiveContainer>
    </div>
  )
}

export default Daily

Daily.propTypes = {
  daily: PropTypes.shape({
    apparentTemperatureHighTime: PropTypes.number,
    apparentTemperatureHigh: PropTypes.number,
    apparentTemperatureLow: PropTypes.number,
    apparentTemperatureLowTime: PropTypes.number,
    apparentTemperatureMax: PropTypes.number,
    apparentTemperatureMaxTime: PropTypes.number,
    apparentTemperatureMin: PropTypes.number,
    apparentTemperatureMinTime: PropTypes.number,
    cloudCover: PropTypes.number,
    dewPoint: PropTypes.number,
    humidity: PropTypes.number,
    icon: PropTypes.string,
    moonPhase: PropTypes.number,
    ozone: PropTypes.number,
    precipIntensity: PropTypes.number,
    precipIntensityMax: PropTypes.number,
    precipIntensityMaxTime: PropTypes.number,
    precipProbability: PropTypes.number,
    pressure: PropTypes.number,
    summary: PropTypes.string,
    sunriseTime: PropTypes.number,
    sunsetTime: PropTypes.number,
    temperatureHigh: PropTypes.number,
    temperatureHighTime: PropTypes.number,
    temperatureLow: PropTypes.number,
    temperatureLowTime: PropTypes.number,
    temperatureMax: PropTypes.number,
    temperatureMaxTime: PropTypes.number,
    temperatureMin: PropTypes.number,
    temperatureMinTime: PropTypes.number,
    time: PropTypes.number,
    uvIndex: PropTypes.number,
    uvIndexTime: PropTypes.number,
    visibility: PropTypes.number,
    windBearing: PropTypes.number,
    windGust: PropTypes.number,
    windGustTime: PropTypes.number,
    windSpeed: PropTypes.number,
  }),

  timezone: PropTypes.string.isRequired,

  dateText: PropTypes.shape({
    weekDay: PropTypes.arrayOf(PropTypes.string),
    today: PropTypes.string.isRequired,
    temperature: PropTypes.string.isRequired,
    temperatureMax: PropTypes.string.isRequired,
    temperatureMin: PropTypes.string.isRequired,
  }),
}


/*
apparentTemperatureHigh: -8.77
apparentTemperatureHighTime: 1543582800
apparentTemperatureLow: -15.57
apparentTemperatureLowTime: 1543640400
apparentTemperatureMax: -8.77
apparentTemperatureMaxTime: 1543582800
apparentTemperatureMin: -13.37
apparentTemperatureMinTime: 1543615200
cloudCover: 0.04
dewPoint: -12.43
humidity: 0.54
icon: "wind"
moonPhase: 0.76
ozone: 285.51
precipIntensity: 0
precipIntensityMax: 0.0025
precipIntensityMaxTime: 1543579200
precipProbability: 0
pressure: 1026.8
summary: "Breezy in the morning."
sunriseTime: 1543559352
sunsetTime: 1543588381
temperatureHigh: -1.95
temperatureHighTime: 1543579200
temperatureLow: -8.79
temperatureLowTime: 1543640400
temperatureMax: -1.95
temperatureMaxTime: 1543579200
temperatureMin: -6.21
temperatureMinTime: 1543615200
time: 1543532400
uvIndex: 1
uvIndexTime: 1543572000
visibility: 16.09
windBearing: 145
windGust: 15.09
windGustTime: 1543561200
windSpeed: 7.16
*/