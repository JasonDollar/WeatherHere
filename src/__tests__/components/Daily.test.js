import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { appTestData as data } from '../../data/fixtures'
import Daily from '../../components/Daily/Daily'


let daily
let dateText
let timezone
let units
let wrapper

beforeAll(() => {
  daily = data.weather.daily
  dateText = data.text.pl.date
  timezone = data.timezone
  units = data.units.si.id
  wrapper = shallow(
    <Daily 
      daily={daily}
      dateText={dateText}
      timezone={timezone}
      units={units}
    />,
  )
})

describe('Daily component test', () => {
  test('Matches snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
  test('proper props given to Summary component', () => {
    const element = wrapper.find('Summary')
    expect(element.props().children).toBe(daily.summary)
  }) 
})