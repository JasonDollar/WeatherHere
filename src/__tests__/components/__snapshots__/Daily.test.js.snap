// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Daily component test Matches snapshot 1`] = `
<section
  className="Daily"
>
  <Summary>
    Dzisiaj i w piątek deszcz, jutro ocieplenie do 19°C.
  </Summary>
  <div
    className="chartContainer"
  >
    <ResponsiveContainer
      debounce={0}
      height={300}
      width="95%"
    >
      <BarChart
        barCategoryGap="10%"
        barGap={4}
        data={
          Array [
            Object {
              "Temperatura (°C)": Array [
                10.1,
                11.2,
              ],
              "name": "Dziś",
              "summary": "Rano deszcz, słaby wiatr.",
            },
            Object {
              "Temperatura (°C)": Array [
                18.9,
                8.6,
              ],
              "name": "Wt",
              "summary": "Rano słaby wiatr, średnie zachmurzenie, ustanie wieczorem.",
            },
            Object {
              "Temperatura (°C)": Array [
                15.8,
                8.5,
              ],
              "name": "Śr",
              "summary": "Rano niewielkie zachmurzenie.",
            },
            Object {
              "Temperatura (°C)": Array [
                18.6,
                13.4,
              ],
              "name": "Czw",
              "summary": "Średnie zachmurzenie w ciągu dnia.",
            },
            Object {
              "Temperatura (°C)": Array [
                16.5,
                8,
              ],
              "name": "Pt",
              "summary": "Rano niewielki deszcz.",
            },
            Object {
              "Temperatura (°C)": Array [
                12.7,
                8.9,
              ],
              "name": "Sob",
              "summary": "Średnie zachmurzenie w ciągu dnia.",
            },
            Object {
              "Temperatura (°C)": Array [
                16.8,
                10.3,
              ],
              "name": "Nie",
              "summary": "Średnie zachmurzenie, ustanie wieczorem.",
            },
            Object {
              "Temperatura (°C)": Array [
                14.4,
                10.6,
              ],
              "name": "Pon",
              "summary": "Średnie zachmurzenie w ciągu dnia.",
            },
          ]
        }
        layout="horizontal"
        margin={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        maxBarSize={60}
        reverseStackOrder={false}
        stackOffset="none"
      >
        <defs>
          <linearGradient
            id="colorPrimary"
            x1="0"
            x2="0"
            y1="0"
            y2="1"
          >
            <stop
              offset="35%"
              stopColor=""
              stopOpacity={1}
            />
            <stop
              offset="95%"
              stopColor=""
              stopOpacity={0.98}
            />
          </linearGradient>
        </defs>
        <CartesianGrid
          fill="none"
          horizontal={true}
          horizontalFill={Array []}
          horizontalPoints={Array []}
          stroke="#ccc"
          strokeDasharray="3 3"
          vertical={true}
          verticalFill={Array []}
          verticalPoints={Array []}
        />
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="name"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          stroke=""
          tickCount={5}
          type="category"
          width={0}
          xAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={0}
          hide={false}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          stroke=""
          tick={
            Object {
              "stroke": "",
            }
          }
          tickCount={5}
          type="number"
          width={60}
          yAxisId={0}
        />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          content={
            <CustomTooltip
              tempText={
                Object {
                  "temperature": "Temperatura",
                  "temperatureMax": "Temperatura Max.",
                  "temperatureMin": "Temperatura Min.",
                  "today": "Dziś",
                  "weekDay": Array [
                    "Niedziela",
                    "Poniedziałek",
                    "Wtorek",
                    "Środa",
                    "Czwartek",
                    "Piątek",
                    "Sobota",
                    "Dziś",
                  ],
                  "weekDayShort": Array [
                    "Nie",
                    "Pon",
                    "Wt",
                    "Śr",
                    "Czw",
                    "Pt",
                    "Sob",
                  ],
                }
              }
              type="daily"
              units="si"
              windowWidth={NaN}
            />
          }
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
        <Legend
          align="center"
          iconSize={14}
          layout="horizontal"
          verticalAlign="bottom"
        />
        <Bar
          animationBegin={0}
          animationDuration={400}
          animationEasing="ease"
          data={Array []}
          dataKey="Temperatura (°C)"
          fill="url(#colorPrimary)"
          hide={false}
          isAnimationActive={true}
          layout="vertical"
          legendType="rect"
          minPointSize={0}
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          unit="si"
          xAxisId={0}
          yAxisId={0}
        />
      </BarChart>
    </ResponsiveContainer>
  </div>
</section>
`;
